ledger.go

```diff
@@ -239,7 +240,14 @@ func (l *Ledger) Close() {
 // is returned if the transaction has already existed int he ledgers graph
 // beforehand.
 func (l *Ledger) AddTransaction(txs ...Transaction) error {
-	l.transactions.BatchAdd(l.blocks.Latest().ID, txs...)
+	block := l.blocks.Latest()
+	for _, tx := range txs {
+		if tx.Sender == l.client.Keys().PublicKey() && tx.Block != block.Index+1 {
```

Comment:
The only check needed is tx.Block >= latestBlock.Index - GetPruningLimit(). To prevent uint64 underflow, tx.Block + GetPruningLimit() >= latestBlock.Index.

The sender doesn't need to be checked.